source/MaterialXRuntime/Private/PvtPrim.h

```diff
@@ -240,31 +247,164 @@ class PvtPrim : public PvtObject
 protected:
     PvtPrim(const RtTypeInfo* typeInfo, const RtToken& name, PvtPrim* parent);
 
-    void addChildPrim(const PvtPrim* prim);
-    void removeChildPrim(const PvtPrim* prim);
+    void addChildPrim(PvtPrim* prim);
+    void removeChildPrim(PvtPrim* prim);
 
     const RtTypeInfo* _typeInfo;
 
     // Relationships
-    PvtDataHandleMap _relMap;
-    PvtDataHandleVec _relOrder;
+    PvtObjectList _rel;
 
-    // Attributes
-    PvtDataHandleMap _attrMap;
-    PvtDataHandleVec _attrOrder;
+    // Inputs
+    PvtObjectList _inputs;
+
+    // Outputs
+    PvtObjectList _outputs;
 
     // Child prims
-    PvtDataHandleMap _primMap;
-    PvtDataHandleVec _primOrder;
+    PvtObjectList _prims;
 
     PvtAllocator _allocator;
 
     friend class PvtApi;
     friend class PvtStage;
     friend class RtNodeGraph;
+    friend class RtPrimIterator;
+    friend class RtInputIterator;
+    friend class RtOutputIterator;
     friend class RtRelationshipIterator;
 };
 
+
+struct PvtAttributeSpec
+{
+    RtToken name;
+    RtToken type;
+//    RtValue value; TODO: Fix me!
```

Comment:
I guess you meant to remove this?

Reply:
It's there as a reminder that I should use RtValue instead of string value later. I need some changes from my codegen change list to fix that, so was planning on fixing it after the PR's are merged.. But I can improve on the TODO text :)